zerolog/logger_test.go

```diff
@@ -0,0 +1,305 @@
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.E SOFTWARE.
+ *
+ * This file may have been modified by CloudWeGo authors. All CloudWeGo
+ * Modifications are Copyright 2022 CloudWeGo Authors.
+ */
+
+package zerolog
+
+import (
+	"bytes"
+	"context"
+	"testing"
+
+	"github.com/cloudwego/hertz/pkg/common/hlog"
+	"github.com/rs/zerolog"
+	"github.com/rs/zerolog/log"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestFrom(t *testing.T) {
+	b := &bytes.Buffer{}
+
+	zl := zerolog.New(b).With().Str("key", "test").Logger()
+	l := From(zl)
+
+	l.Info("foo")
+
+	assert.Equal(
+		t,
+		`{"level":"info","key":"test","message":"foo"}
+`,
+		b.String(),
+	)
+}
+
+func TestUnwrap(t *testing.T) {
+	l := New()
+
+	logger := l.Unwrap()
+
+	assert.NotNil(t, logger)
+	assert.IsType(t, &zerolog.Logger{}, logger)
+}
+
+func TestLog(t *testing.T) {
+	b := &bytes.Buffer{}
+	l := New()
+	l.SetOutput(b)
+
+	l.Trace("foo")
+	assert.Equal(
+		t,
+		`{"level":"debug","message":"foo"}
+`,
+		b.String(),
+	)
+
+	b.Reset()
+	l.Debug("foo")
+	assert.Equal(
+		t,
+		`{"level":"debug","message":"foo"}
+`,
+		b.String(),
+	)
+
+	b.Reset()
+	l.Info("foo")
+	assert.Equal(
+		t,
+		`{"level":"info","message":"foo"}
+`,
+		b.String(),
+	)
+
+	b.Reset()
+	l.Notice("foo")
+	assert.Equal(
+		t,
+		`{"level":"warn","message":"foo"}
+`,
+		b.String(),
+	)
+
+	b.Reset()
+	l.Warn("foo")
+	assert.Equal(
+		t,
+		`{"level":"warn","message":"foo"}
+`,
+		b.String(),
+	)
+
+	b.Reset()
+	l.Error("foo")
+	assert.Equal(
+		t,
+		`{"level":"error","message":"foo"}
+`,
+		b.String(),
+	)
+}
+
+func TestLogf(t *testing.T) {
+	b := &bytes.Buffer{}
+	l := New()
+	l.SetOutput(b)
+
+	l.Tracef("foo%s", "bar")
+	assert.Equal(
+		t,
+		`{"level":"debug","message":"foobar"}
+`,
+		b.String(),
+	)
+
+	b.Reset()
+	l.Debugf("foo%s", "bar")
+	assert.Equal(
+		t,
+		`{"level":"debug","message":"foobar"}
+`,
+		b.String(),
+	)
+
+	b.Reset()
+	l.Infof("foo%s", "bar")
+	assert.Equal(
+		t,
+		`{"level":"info","message":"foobar"}
+`,
+		b.String(),
+	)
+
+	b.Reset()
+	l.Noticef("foo%s", "bar")
+	assert.Equal(
+		t,
+		`{"level":"warn","message":"foobar"}
+`,
+		b.String(),
+	)
+
+	b.Reset()
+	l.Warnf("foo%s", "bar")
+	assert.Equal(
+		t,
+		`{"level":"warn","message":"foobar"}
+`,
+		b.String(),
+	)
+
+	b.Reset()
+	l.Errorf("foo%s", "bar")
+	assert.Equal(
+		t,
+		`{"level":"error","message":"foobar"}
+`,
+		b.String(),
+	)
+}
+
+func TestCtxTracef(t *testing.T) {
+	b := &bytes.Buffer{}
+	l := New()
+	l.SetOutput(b)
+	ctx := l.log.WithContext(context.Background())
```

Comment:
This should be mentioned in readme. 
// If no logger is associated, DefaultContextLogger is returned, unless DefaultContextLogger is nil, in which case a disabled logger is returned.

Reply:
Fixed